cartoframes/data/enrichment/enrichment_service.py

```diff
@@ -111,38 +152,77 @@ def __process_filters(filters_dict):
+
+        if variable_name not in agg_operators_result:
+            agg_operators_result[variable_name] = variable.agg_method
+
+    return agg_operators_result
 
-    table_to_variables = __process_enrichment_variables(variables_id)
-    table_data_enrichment = list(table_to_variables.keys()).pop()
-    table_geo_enrichment = __get_name_geotable_from_datatable(table_data_enrichment)
-    variables_list = list(table_to_variables.values()).pop()
 
-    return table_data_enrichment, table_geo_enrichment, variables_list
+def __get_tables_and_variables(variables, user_dataset):
 
+    table_to_geotable, table_to_variables, table_to_project, table_to_dataset =\
+        __process_enrichment_variables(variables, user_dataset)
 
-def __process_enrichment_variables(variables):
+    return table_to_geotable, table_to_variables, table_to_project, table_to_dataset
+
+
+def __process_enrichment_variables(variables_id, user_dataset):
+    table_to_geotable = dict()
     table_to_variables = defaultdict(list)
+    table_to_project = dict()
+    table_to_dataset = dict()
 
-    for variable in variables:
-        variable_split = variable.split('.')
-        table, variable = variable_split[-2], variable_split[-1]
+    for variable_id in variables_id:
+        variable_split = variable_id.id.split('.')
+        project, dataset, table, variable = variable_split
+
+        if project != _PUBLIC_PROJECT:
+            table = '{dataset}_{table}'.format(dataset=dataset,
+                                               table=table,
+                                               user_dataset=user_dataset)
+
+        if table not in table_to_dataset:
+            if project != _PUBLIC_PROJECT:
+                table_to_dataset[table] = user_dataset
+            else:
+                table_to_dataset[table] = _PUBLIC_DATASET
+
+        if table not in table_to_geotable:
+            geotable = __get_name_geotable_from_datatable(table)
+
+            if project != _PUBLIC_PROJECT:
+                geotable = '{dataset}_{geotable}'.format(dataset=dataset,
+                                                         geotable=geotable,
+                                                         user_dataset=user_dataset)
+
+            table_to_geotable[table] = geotable
+
+        if table not in table_to_project:
+            if project == _PUBLIC_PROJECT:
+                table_to_project[table] = _PUBLIC_PROJECT
+            else:
+                table_to_project[table] = _WORKING_PROJECT
 
         table_to_variables[table].append(variable)
 
-    return table_to_variables
+    return table_to_geotable, table_to_variables, table_to_project, table_to_dataset
 
 
 def __get_name_geotable_from_datatable(datatable):
+
     datatable_split = datatable.split('_')
-    geo_information = datatable_split[2:5]
+
+    if len(datatable_split) == 8:
+        geo_information = datatable_split[3:6]
+    elif len(datatable_split) == 7:
+        geo_information = datatable_split[2:5]
```

Comment:
This should be responsibility of the Dataset class of the catalog (beware the magic numbers BTW).

Reply:
I would create another issue for this, we have to think this well, because this is logic due to dataset licensing, so we have to add the provider name in the licensed table, but if it is public data so doesn't.

Reply:
Refer to https://github.com/CartoDB/data-observatory/issues/238